operator/api/v1beta1/lokistack_types.go

```diff
@@ -0,0 +1,669 @@
+	ConditionFailed LokiStackConditionType = "Failed"
+
+	// ConditionDegraded defines the condition that some or all components in the Loki deployment
+	// are degraded or the cluster cannot connect to object storage.
+	ConditionDegraded LokiStackConditionType = "Degraded"
+)
+
+// LokiStackConditionReason defines the type for valid reasons of a Loki deployment conditions.
+type LokiStackConditionReason string
+
+const (
+	// ReasonFailedComponents when all/some LokiStack components fail to roll out.
+	ReasonFailedComponents LokiStackConditionReason = "FailedComponents"
+	// ReasonPendingComponents when all/some LokiStack components pending dependencies
+	ReasonPendingComponents LokiStackConditionReason = "PendingComponents"
+	// ReasonReadyComponents when all LokiStack components are ready to serve traffic.
+	ReasonReadyComponents LokiStackConditionReason = "ReadyComponents"
+	// ReasonMissingObjectStorageSecret when the required secret to store logs to object
+	// storage is missing.
+	ReasonMissingObjectStorageSecret LokiStackConditionReason = "MissingObjectStorageSecret"
+	// ReasonInvalidObjectStorageSecret when the format of the secret is invalid.
+	ReasonInvalidObjectStorageSecret LokiStackConditionReason = "InvalidObjectStorageSecret"
+	// ReasonInvalidReplicationConfiguration when the configurated replication factor is not valid
+	// with the select cluster size.
+	ReasonInvalidReplicationConfiguration LokiStackConditionReason = "InvalidReplicationConfiguration"
+	// ReasonMissingGatewayTenantSecret when the required tenant secret
+	// for authentication is missing.
+	ReasonMissingGatewayTenantSecret LokiStackConditionReason = "MissingGatewayTenantSecret"
+	// ReasonInvalidGatewayTenantSecret when the format of the secret is invalid.
+	ReasonInvalidGatewayTenantSecret LokiStackConditionReason = "InvalidGatewayTenantSecret"
+	// ReasonInvalidTenantsConfiguration when the tenant configuration provided is invalid.
+	ReasonInvalidTenantsConfiguration LokiStackConditionReason = "InvalidTenantsConfiguration"
+	// ReasonMissingGatewayOpenShiftBaseDomain when the reconciler cannot lookup the OpenShift DNS base domain.
+	ReasonMissingGatewayOpenShiftBaseDomain LokiStackConditionReason = "MissingGatewayOpenShiftBaseDomain"
+)
+
+// PodStatusMap defines the type for mapping pod status to pod name.
+type PodStatusMap map[corev1.PodPhase][]string
+
+// LokiStackComponentStatus defines the map of per pod status per LokiStack component.
+// Each component is represented by a separate map of v1.Phase to a list of pods.
+type LokiStackComponentStatus struct {
+	// Compactor is a map to the pod status of the compactor pod.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=status,xDescriptors="urn:alm:descriptor:com.tectonic.ui:podStatuses",displayName="Compactor",order=5
+	Compactor PodStatusMap `json:"compactor,omitempty"`
```

Comment:
The index gateway seems missing in that status

Reply:
Solved in: https://github.com/grafana/loki/pull/4945/commits/caa0b7f2eae95e6e372dbc147fed44d15f2818df